### PR TITLE
[WIP] docs: use sandpack for demo

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
image

由于 builtin 在 markdown 中使用局限性比较高，所以封装了一下，默认包含了 antd 和 react 相关依赖，只需要写 demo 即可。
在 markdown 中这样写就会变成一个 sandpack demo：

````
```sandpack
// export 一个组件
export default () => {};
```
````

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |     -      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 4d7ae73</samp>

Added a new `Sandpack` component to render interactive code examples in the documentation. Replaced the iframe codesandbox demos with the `Sandpack` component in the `getting-started` documents. Updated the `rehypeAntd` function and the `package.json` file to support the new component.

### <samp>🤖 Generated by Copilot at 4d7ae73</samp>

*  Add `@codesandbox/sandpack-react` dependency to `package.json` for interactive code demos ([link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R165))
*  Create `Sandpack` component by re-exporting from `@codesandbox/sandpack-react` in `.dumi/theme/builtins/Sandpack/Sandpack.ts` ([link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-31fc6f55d282bb5636fe9529aba3b0bccda9d34a9b79cba953d723d2c7a6a221R1-R3))
*  Customize `Sandpack` component with lazy loading and sandbox configuration in `.dumi/theme/builtins/Sandpack/index.tsx` ([link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-eb1983e6c2de0da4944ef9eb6428a1850e347a11b36c1b8f18dfacada229c76cR1-R45))
*  Replace `SourceCode` nodes with `Sandpack` nodes in `rehypeAntd` function in `.dumi/rehypeAntd.ts` if language is `sandpack` ([link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-f0d95b6f380e7a072830e7026f73450a2056d161a2b265783fa556338e677525L11-R11), [link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-f0d95b6f380e7a072830e7026f73450a2056d161a2b265783fa556338e677525R69-R82))
*  Replace iframe elements with `Sandpack` components and add source code blocks with `sandpack` language in `docs/react/getting-started.en-US.md` and `docs/react/getting-started.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-42716025870d97fefe509fc56761684f23c617df27964d33a2a44b8eb51a8a38L18-R18), [link](https://github.com/ant-design/ant-design/pull/43854/files?diff=unified&w=0#diff-30aba60ae4a347b7452e2a4a66635f4be7e498b360267b963f87c35772c92aaeL16-R32))

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

File: .dumi/theme/builtins/Sandpack/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const OriginSandpack = React.lazy(() => import('./Sandpack'));
+
+const Sandpack = ({ children }: { children: ReactNode }) => {
+  const [searchParams] = useSearchParams();
+
+  return (
+    <Suspense fallback="loading">
+      <OriginSandpack
+        theme={searchParams.getAll('theme').includes('dark') ? 'dark' : undefined}
+        customSetup={{
```
Comment (li-jia-nan) [inline .dumi/theme/builtins/Sandpack/index.tsx:14]: 这个配置里面可以加个 `"name": "antd-demo",`

Comment (li-jia-nan) [reply .dumi/theme/builtins/Sandpack/index.tsx:14]: 另外感觉这个配置可以抽出去单独定义一下？

Comment (MadCcc) [reply .dumi/theme/builtins/Sandpack/index.tsx:14]: 这是个 builtin，以后直接在 md 里写代码就好了

Comment (MadCcc) [reply .dumi/theme/builtins/Sandpack/index.tsx:14]: https://github.com/ant-design/ant-design/pull/43854/files#diff-42716025870d97fefe509fc56761684f23c617df27964d33a2a44b8eb51a8a38R18
像这样

**Comment by MadCcc** [comment]
注水有问题，先 WIP

**Comment by li-jia-nan** [comment]
WIP 状态直接合了嘛？

**Comment by yoyo837** [comment]
关小黑屋.

**Pull request merged**